tests/run-pass/fs.rs

```diff
@@ -182,6 +183,18 @@ fn test_file_set_len() {
     remove_file(&path).unwrap();
 }
 
+fn test_file_sync() {
+    let bytes = b"Hello, World!\n";
+    let path = prepare_with_content("miri_test_fs_sync.txt", bytes);
+
+    // Test that we can call sync_data and sync_all (can't readily test effects of this operation)
+    let file = OpenOptions::new().write(true).open(&path).unwrap();
```

Comment:
Please also test read-only files. Currently you are not even testing the new Windows-only codepath, are you?